packages/mcdoc/src/type/index.ts

```diff
@@ -17,6 +17,17 @@ export type NumericRange = {
 	min?: number
 	max?: number
 }
+export namespace NumericRange {
+	export function isInRange(range: NumericRange, val: number): boolean {
+		if (range.min && (((range.kind & 0b10) > 1 && val <= range.min) || val < range.min)) {
+			return false;
+		}
+		if (range.max && (((range.kind & 0b01) > 1 && val >= range.max) || val > range.max)) {
+			return false;
+		}
```

Comment:
i think the checks won't work properly if `range.min` or `range.max` is `0`. we probably need the full `range.min !== undefined` check.

and i personally think it'd be clearer in intent if the check is written using a ternary operator: `(range.kind & 0b10) ? val <= range.min : val < range.min`. we could even write a help function `RangeKind.isLeftExclutive(kind)` to replace `(range.kind & 0b10)`.

Reply:
This is done.